prov/gni/src/gnix_util.c

```diff
@@ -191,3 +193,98 @@ int gnixu_to_fi_errno(int err)
 	else
 		return -FI_EOTHER;
 }
+
+/* Indicate that the next task spawned will be restricted to cores assigned to
+ * corespec. */
+int _gnix_task_is_not_app(void)
+{
+	size_t count;
+	int fd;
+	char *filename;
+	int rc = 0;
+	char val_str[] = "0";
+	int val_str_len = strlen(val_str);
+
+	filename = malloc(PATH_MAX);
```

Comment:
Why not use a stack variable for filename? It's size is known beforehand and it's lifetime is limited to the function. Are you worried about exhausting the stack?